src/plugins/vis_builder/public/application/components/right_nav.tsx

```diff
@@ -109,3 +109,5 @@ const OptionItem = ({ icon, title }: { icon: IconType; title: string }) => (
     <span>{title}</span>
   </>
 );
+
+export const RightNav = React.memo(RightNavUI);
```

Comment:
This fixes #3134. The issue was a race condition that caused the state to be reset just as soon as it was set because of the resize observer for OUI's resizable component